Core need some love.

FrankFFFF opened the pull request:
Keep reading and get familiar with it. :neckbeard:

ADmad commented:
Seems you went a bit overboard with your love. Your changes have caused failing tests.

Fieah commented:
LOL, will check.

ADmad commented on src/Core/Configure/FileConfigTrait.php:
```diff
@@ -46,14 +46,7 @@ protected function _getFilePath($key, $checkExists = false)
-        }
-
-        $file .= $this->_extension;
+        $file = ($plugin ? Plugin::configPath($plugin) : $this->_path) . $key . $this->_extension;
```
Using ternary and concatenation in single statement is not good for readability. Keep the concatenation separate.

Fieah replied:
OK, will revert this.

Fieah replied:
Hmm...6 lines to 1 lines with spaces.
I'm still wondering how bad this can be. :8ball:

ADmad commented on src/Core/functions.php:
```diff
@@ -153,13 +153,9 @@ function pr($var)
      */
     function pj($var)
     {
-        if (!Configure::read('debug')) {
```
This should not be removed. Throughout the core we try and return early from method when possible.

Fieah replied:
Actually this is same with `pr()` , so i chose to follow it.

ADmad replied:
Then it would be better to change `pr()` instead.

Fieah replied:
OK.
I'm not sure what caused this to break, should i make another pull request or just change in here?

ADmad replied:
You can make change to this PR itself and at the end squash commits.

antograssiot commented on src/Core/StaticConfigTrait.php:
```diff
@@ -72,15 +72,18 @@
-                static::config($name, $settings);
+        if ($config === null) {
+			// Read config.
+			if (is_string($key)) {
```
indentations seems of (should use space)

Fieah replied:
Hi, had noticed this just now, thanks :)

Fieah commented:
`FILE: /home/travis/build/cakephp/cakephp/src/Core/functions.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 141 | ERROR | Expected 0 blank lines before closing function brace;`
This gotta be kidding me, since when this became an issue?

Fieah commented:
`zend_mm_heap corrupted`
This is not related, right? Is there another way to make it re-run?
Thanks.

AD7six commented:
@Fieah what's "it"? Depends whether it's persistent as to whether it's _probably_ related or not.

Any commit/change will start a new build, team members can them (as I just did for the sqlite run).

Fieah commented:
@AD7six  Hi,
Yeah, i mean re-run Travis, now everything seems to be working fine, thank you :)

rchavik commented:
Umm,  it's good that love is all around. But what is this change trying to improve?

AD7six commented on src/Core/functions.php:
```diff
@@ -156,11 +158,9 @@ function pj($var)
-            printf('<pre class="pj">%s</pre>', trim(json_encode($var, JSON_PRETTY_PRINT)));
-        }
+
+        $template = PHP_SAPI !== 'cli' ? '<pre class="pr">%s</pre>' : "\n%s\n\n";
```
the pre class name has changed here

Fieah replied:
Nice catch, another boo boo made by me. 
Thank you ;)

rchavik commented:
If it's not clear, I'm not opposed to the changes.  But I hope you'll amend the commit message to explain what a particular commit does.  The current ones are amusing easter eggs, but ... :)

AD7six commented:
@Fieah when you're happy with the result, can you squash your changes into one commit giving it a descriptive commit title and description please.

:+1: otherwise.

Fieah commented:
@rchavik  @AD7six 
Yeah, will make another PR later, this one is messy.
Thanks for the help :)

AD7six commented:
@Fieah you don't need to create a new pr.

    git fetch
    git rebase origin/master # read 
    git push -f

and reopen this PR.

Fieah commented:
@AD7six 
Thank you, will try. :+1:

thaJeztah commented:
@Fieah instructions above are correct, but you need to **first** reopen the PR, **then** `git push -f`, because GitHub will not update a PR if it's closed

Fieah commented:
Hi @thaJeztah 
OK, trying now, thank you. :)

markstory commented:
Changes all seem reasonable. I personally find less deeply nested control structures easier to work with, but it does result in some duplicated conditions.

thaJeztah commented on src/Core/StaticConfigTrait.php:
```diff
@@ -72,15 +72,18 @@
-            return;
         }
 
         if (isset(static::$_config[$key])) {
```
Not related to this PR, but this will break if `$key` is an array?

markstory replied:
Yes if config is not null and key is an array bad things will happen.

FrankFFFF committed:
Core need some love.

1. Core/Plugin.php: load(): Don't declare $pluginPath in loop.
2. Standardize the code block of pr() and pj()
3. Simple structures use ternary instead of if/else

Fieah commented:
FINALLY, commits DONE, ci PASS.
Thanks a lot guys, i have learned something new today. :raised_hands:
